### ISSUE TITLE
Support for adaptive icons

### ISSUE THREAD

**Issue by UnderSampled** [dkanada/frost#229]
Though not all, a great many icons in this pack are a transparent silhouette/stencil cut out of a solid shape. There is no consistency on which shape is used, and the choice makes no difference to the representation of the app. These would be great candidates for "Adaptive Icons", available since Android 8: https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive
This would also bring cleaner design uniformity to these icons without hurting usability. Other icons rely on their outlines to be differentiated (important with the lack of color), so this is probably not practical to require for all icons, but it could be encouraged for new designs when it is possible and doesn't hurt the experience.

Examples:
Circle: Discord, NewPipe, K-9 Mail, Spotify, LibreTorrent
Square (rounded): PayPal, Forcastie, Calendar, Google Maps, Flow Free, Puzzles, MuPDF
Rectangle: Contacts, SatStat

**Comment by dkanada** [comment]
I'm not sure adaptive icons work with transparent space inside the icon, which is what 90% of the icons in this pack use.

**Comment by UnderSampled** [comment]
Oh. It seems that any transparency in the background of an adaptive icons is rendered simply as black. I was imagining it just set a mask over a square icon that could still have transparency inside.

Perhaps we could implement our own run-time masking based on a user setting which could match the available masks (circle, square, etc) used by common launchers.

We could at least unify the outline shape (i.e. choose circle or square). Maybe still leave it as an unmasked square (or define the inner stencil shape only) in the svg source and apply the mask at render time, in case we want to develop this feature in the future.

One thing this can do which wouldn't be possible otherwise is the option to take what is currently only the negative stencil and use it as the positive icon (no outer shape).

**Comment by Donnnno** [comment]
When I was designing a lot of icons lately, I would prefer round shapes. But sometimes it just doesn't work for the icon in it and it can look kinda ugly, so that's why I sometimes choose other shapes now.

**Comment by dkanada** [comment]
Yeah, I don't think this icon pack will be switching to a single shape any time soon. We can't accomplish it with adaptive icons and there are many icons that wouldn't look good with just the inner shape either.

**Comment by ohemelaar** [comment]
Could there be a way to provide an icon with different shapes (base icon, transparent in square, transparent in circle)? This could allow the user to have a more unified style, while still letting them chose.

I don't know the details of the android implementation side, but I've tested by hand with some SVG files and it's usually just a matter of removing a single part of a path to go from transparent to base icon, or adding a circle/square in the path for the other way around. It seems feasible to support all three styles for most icons without having to draw everything from scratch again.
